File: tests/fixtures/ruby_library/library_v2/lib/mylib.rb
# MyLib v2.0.0 - A sample Ruby library for demonstrating API upgrades.
#
# Breaking changes from v1.0.0:
# - User renamed to UserAccount with new email attribute
# - Status renamed to ConnectionStatus
# - get_user renamed to fetch_user
# - process_data renamed to transform_data
# - connect now requires port and timeout parameters
# - save no longer takes sync parameter
# - query parameters reordered from (a, b, c) to (c, a, b)
# - find now returns nil instead of raising for empty keys
# - parse now returns ParseResult instead of string
# - deprecated_fn has been removed
# - Utils renamed to Helpers

module MyLib
  # A user account in the system (renamed from User).
  class UserAccount
    attr_reader :id, :name, :email

    def initialize(id, name, email)
      @id = id
      @name = name
      @email = email
    end

    def to_s
      "UserAccount{id=#{@id}, name='#{@name}', email='#{@email}'}"
    end
  end

  # Connection status (renamed from Status).
  module ConnectionStatus
    CONNECTED = :connected
    DISCONNECTED = :disconnected
    ERROR = :error
  end

  # Configuration for the library (with new port attribute).
  class Config
    attr_reader :host, :port

    def initialize(host, port)
      @host = host
      @port = port
    end
  end

  # Database query result item.
  class Item
    attr_reader :key, :value

    def initialize(key, value)
      @key = key
      @value = value
    end
  end

  # Result of parsing operations (new in v2).
  class ParseResult
    attr_reader :output, :warnings

    def initialize(output, warnings = [])
      @output = output
      @warnings = warnings
    end
  end

  class << self
    # Fetch a user by their ID (renamed from get_user).
    def fetch_user(id)
      return nil if id <= 0
      UserAccount.new(id, "User#{id}", "user#{id}@example.com")
    end

    # Transform raw data bytes (renamed from process_data).
    def transform_data(data)
      data.bytes.map { |b| (b + 1) % 256 }
    end

    # Connect to a host with port and timeout (signature changed).
    def connect(host, port, timeout)
      raise ArgumentError, "empty host" if host.nil? || host.empty?
      raise ArgumentError, "invalid port" if port <= 0
      ConnectionStatus::CONNECTED
    end

    # Save data (sync parameter removed).
    def save(data)
      puts "Saved: #{data}"
    end

    # Query with reordered parameters (c, a, b instead of a, b, c).
    def query(c, a, b)
      results = []
      results << Item.new(a, b.to_s) if c
      results
    end

    # Find an item by key (now returns nil for empty keys).
    def find(key)
      return nil if key.nil? || key.empty?
      Item.new(key, "found")
    end

    # Parse a string into a ParseResult (return type changed).
    def parse(input)
      if input.nil? || input.empty?
        ParseResult.new("", ["empty input"])
      else
        ParseResult.new(input.upcase)
      end
    end

    # Note: deprecated_fn has been removed in v2
  end

  # Helper functions (renamed from Utils).
  module Helpers
    class << self
      # A helper function that returns a greeting.
      def helper
        "Hello from helpers"
      end

      # Format a user account for display.
      def format_user(user)
        "UserAccount(id=#{user.id}, name=#{user.name}, email=#{user.email})"
      end

      # Create a config map.
      def create_config_map
        { "version" => "2.0.0" }
      end
    end
  end
end

File: tests/fixtures/csharp_library/library_v2/MyLib.cs
// MyLib v2.0.0 - A sample C# library for demonstrating API upgrades.
//
// Breaking changes from v1.0.0:
// - User renamed to UserAccount with new Email property
// - Status renamed to ConnectionStatus
// - GetUser renamed to FetchUser
// - ProcessData renamed to TransformData
// - Connect now requires port and timeout parameters
// - Save no longer takes sync parameter
// - Query parameters reordered from (a, b, c) to (c, a, b)
// - Find now returns Item? instead of Item
// - Parse now returns ParseResult instead of string
// - DeprecatedFn has been removed
// - Utils renamed to Helpers

using System;
using System.Collections.Generic;

namespace MyLib
{
    /// <summary>A user account in the system (renamed from User).</summary>
    public class UserAccount
    {
        public long Id { get; }
        public string Name { get; }
        public string Email { get; } // New property in v2

        public UserAccount(long id, string name, string email)
        {
            Id = id;
            Name = name;
            Email = email;
        }

        public override string ToString() => $"UserAccount{{Id={Id}, Name='{Name}', Email='{Email}'}}";
    }

    /// <summary>Connection status (renamed from Status).</summary>
    public enum ConnectionStatus
    {
        Connected,
        Disconnected,
        Error
    }

    /// <summary>Configuration for the library (with new Port property).</summary>
    public class Config
    {
        public string Host { get; }
        public int Port { get; } // New property in v2

        public Config(string host, int port)
        {
            Host = host;
            Port = port;
        }
    }

    /// <summary>Database query result item.</summary>
    public class Item
    {
        public string Key { get; }
        public string Value { get; }

        public Item(string key, string value)
        {
            Key = key;
            Value = value;
        }
    }

    /// <summary>Result of parsing operations (new in v2).</summary>
    public class ParseResult
    {
        public string Output { get; }
        public List<string> Warnings { get; }

        public ParseResult(string output, List<string> warnings)
        {
            Output = output;
            Warnings = warnings;
        }
    }

    /// <summary>Main library class.</summary>
    public static class Library
    {
        /// <summary>Fetch a user by their ID (renamed from GetUser).</summary>
        public static UserAccount? FetchUser(long id)
        {
            if (id > 0)
            {
                return new UserAccount(id, $"User{id}", $"user{id}@example.com");
            }
            return null;
        }

        /// <summary>Transform raw data bytes (renamed from ProcessData).</summary>
        public static byte[] TransformData(byte[] data)
        {
            var result = new byte[data.Length];
            for (int i = 0; i < data.Length; i++)
            {
                result[i] = (byte)(data[i] + 1);
            }
            return result;
        }

        /// <summary>Connect to a host with port and timeout (signature changed).</summary>
        public static ConnectionStatus Connect(string host, int port, TimeSpan timeout)
        {
            if (string.IsNullOrEmpty(host))
            {
                throw new ArgumentException("empty host");
            }
            if (port <= 0)
            {
                throw new ArgumentException("invalid port");
            }
            return ConnectionStatus.Connected;
        }

        /// <summary>Save data (sync parameter removed).</summary>
        public static void Save(string data)
        {
            Console.WriteLine($"Saved: {data}");
        }

        /// <summary>Query with reordered parameters (c, a, b instead of a, b, c).</summary>
        public static List<Item> Query(bool c, string a, int b)
        {
            var results = new List<Item>();
            if (c)
            {
                results.Add(new Item(a, b.ToString()));
            }
            return results;
        }

        /// <summary>Find an item by key (now returns Item?).</summary>
        public static Item? Find(string key)
        {
            if (string.IsNullOrEmpty(key))
            {
                return null;
            }
            return new Item(key, "found");
        }

        /// <summary>Parse a string into a ParseResult (return type changed).</summary>
        public static ParseResult Parse(string input)
        {
            if (string.IsNullOrEmpty(input))
            {
                return new ParseResult("", new List<string> { "empty input" });
            }
            return new ParseResult(input.ToUpper(), new List<string>());
        }

        // Note: DeprecatedFn has been removed in v2
    }

    /// <summary>Helper functions (renamed from Utils).</summary>
    public static class Helpers
    {
        /// <summary>A helper function that returns a greeting.</summary>
        public static string Helper()
        {
            return "Hello from helpers";
        }

        /// <summary>Format a user account for display.</summary>
        public static string FormatUser(UserAccount user)
        {
            return $"UserAccount(id={user.Id}, name={user.Name}, email={user.Email})";
        }

        /// <summary>Create a config map.</summary>
        public static Dictionary<string, string> CreateConfigMap()
        {
            return new Dictionary<string, string> { { "version", "2.0.0" } };
        }
    }
}
